Data/Binary/Serialise/CBOR/Class.hs

```diff
@@ -204,6 +207,107 @@ instance Serialise BS.ByteString where
     encode = encodeBytes
     decode = decodeBytes
 
+instance Serialise a => Serialise (Const a b) where
+    encode (Const a) = encode a
+    decode = Const <$> decode
+
+instance Serialise a => Serialise (ZipList a) where
+    encode (ZipList xs) = encode xs
+    decode = ZipList <$> decode
+
+instance (Serialise a, Integral a) => Serialise (Ratio a) where
+    encode a = encodeListLen 2
+            <> encode (numerator a)
+            <> encode (denominator a)
+    decode = do decodeListLenOf 2
+                !a <- decode
+                !b <- decode
+                return $ a % b
+
+instance Serialise a => Serialise (Complex a) where
+    encode (r :+ i) = encodeListLen 2
+                   <> encode r
+                   <> encode i
+    decode = do decodeListLenOf 2
+                !r <- decode
+                !i <- decode
+                return $ r :+ i
+
+instance Serialise Ordering where
+    encode a = encodeListLen 1
```

Comment:
It just follows convention for encoding ADTs. They're encoded as list [constructor tag, field1, ...]. I'm not sure if it's documented anywhere. 

Sure it's possible to use more compact I'm not sure whether it worth it